File: backend/classwatch-backend/src/main/java/com/classwatch/backend/model/Lecture.java
package com.classwatch.backend.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Lob;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "lectures")
public class Lecture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String titulo;

    @Column(length = 500)
    private String descricao;

    @Column(name = "audio_path", nullable = false, length = 255)
    private String audioPath;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String transcricao;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String resumo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 32)
    private LectureStatus status = LectureStatus.RECEBIDO;

    @Column(name = "error_message", length = 1000)
    private String errorMessage;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "processing_started_at")
    private Instant processingStartedAt;

    @Column(name = "processing_finished_at")
    private Instant processingFinishedAt;

    @PrePersist
    void prePersist() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void preUpdate() {
        updatedAt = Instant.now();
    }
}

File: mobile/types/lecture.ts
export type LectureStatus = 'RECEBIDO' | 'PROCESSANDO' | 'FINALIZADO' | 'ERRO';

export interface Lecture {
  id: number;
  titulo: string;
  descricao: string;
  transcricao: string | null;
  resumo: string | null;
  status: LectureStatus;
  errorMessage: string | null;
  createdAt: string;
  updatedAt: string;
  processingStartedAt: string | null;
  processingFinishedAt: string | null;
}

export const statusLabels: Record<LectureStatus, string> = {
  RECEBIDO: 'Na fila',
  PROCESSANDO: 'Processando',
  FINALIZADO: 'Concluída',
  ERRO: 'Falha',
};

export function isTerminalStatus(status: LectureStatus) {
  return status === 'FINALIZADO' || status === 'ERRO';
}
